src/EnergyPlus/BaseboardRadiator.cc

```diff
@@ -842,10 +842,11 @@ namespace BaseboardRadiator {
 
                             if (LoadMet > DesCoilLoad) { // if the load met is greater than design load, OK to iterate on UA
                                 // Invert the baseboard model: given the design inlet conditions and the design load, find the design UA.
-                                auto f = [&state, &baseboardNum, DesCoilLoad](Real64 UA) {
+                                auto f = [&state, baseboardNum, DesCoilLoad](Real64 UA) {
                                     state.dataBaseboardRadiator->baseboards(baseboardNum).UA = UA;
+                                    int localBaseBoardNum = baseboardNum;
```

Comment:
Created a local variable that can be passed as a reference to `SimHWConvective`, allowing the lambda capture to capture by value.  There may still be optimization opportunities here, but this should be good for now I think.